src/Loop.php

```diff
@@ -42,12 +42,11 @@ public static function setFactory(DriverFactory $factory = null)
         }
 
         self::$factory = $factory;
+        self::$driver = null; // reset it here, it will be actually instantiated inside execute() or get()
 
         if ($factory === null) {
-            self::$driver = null;
             self::$registry = null;
         } else {
-            self::$driver = self::createDriver();
             self::$registry = [];
```

Comment:
We'd need a getter for the complete registry state. I really don't like that. Stupid `wait` ...